### PR TITLE
Added network-unavailable tolerations when hostNetwork=true.

### LINKED ISSUE #61050: [test failed] gci-gce-alpha-features
https://k8s-testgrid.appspot.com/sig-release-master-blocking#gci-gce-alpha-features
testgrid diff for commit range: https://github.com/kubernetes/kubernetes/compare/f7d9fb474...f7aafaeb4

looks like #59862 

/priority failing-test
/priority critical-urgent
/kind bug
/status approved-for-milestone
/sig apps
/milestone v1.10

cc @jdumars @jberkus 

/assign @k82cn @janetkuo

### PR DESCRIPTION
**Pull request by k82cn**
Signed-off-by: Da K. Ma <klaus1982.cn@gmail.com>

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #61050 

**Release note**:
```release-note
None
```

### CODE REVIEW

**Comment by k82cn** [comment]
According to `hostNetwork: true` in yaml, metadata-proxy will not dependent on container network; when `TaintNodesByCondition` and `ScheduleDaemonSetPods` enabled, metadata-proxy should tolerate `node.kubernetes.io/network-unavailable` taints.

**Comment by krzyzacy** [comment]
cc @jdumars @jberkus

**Comment by krzyzacy** [comment]
/assign @Q-Lee @cjcullen @mikedanese 
someone can review/approve this?

File: pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -832,8 +832,8 @@ func podMatchesNodeLabels(pod *v1.Pod, node *v1.Node) bool {
 			glog.V(10).Infof("Match for RequiredDuringSchedulingIgnoredDuringExecution node selector terms %+v", nodeSelectorTerms)
 			nodeAffinityMatches = nodeAffinityMatches && nodeMatchesNodeSelectorTerms(node, nodeSelectorTerms)
 		}
-
```
Comment (dims) [inline pkg/scheduler/algorithm/predicates/predicates.go]: unrelated whitespace change?

**Comment by k82cn** [comment]
Open another PR (https://github.com/kubernetes/kubernetes/pull/61313) for this which is better 👍

**Comment by jdumars** [comment]
@k82cn if this PR is not needed in favor of #61313 could you please close it?

**Comment by k82cn** [comment]
Because of Pod version issue, can not get `hostNetwork` in admission controller; so this PR is safer for 1.10.

**Comment by k82cn** [comment]
/close
Handled in  #61313 ; sorry for the confusion.

**Comment by k82cn** [comment]
Slack with liggitt@, we

- prefer to change yaml file (https://github.com/kubernetes/kubernetes/pull/61311) instead of depending an optional admission controller (https://github.com/kubernetes/kubernetes/pull/61313)
- I'll update doc (https://github.com/kubernetes/website/pull/7678) about that: Pod using hostNetwork should tolerate NetworkUnavailable taints
- I'll update release note for it

File: cluster/addons/metadata-proxy/gce/metadata-proxy.yaml [hunk 2]
```diff
@@ -37,6 +37,10 @@ spec:
       serviceAccountName: metadata-proxy
       hostNetwork: true
       dnsPolicy: Default
+      tolerations:
```
Comment (mikedanese) [inline cluster/addons/metadata-proxy/gce/metadata-proxy.yaml]: since this is a critical addon should we give it:
```
  tolerations:
  - operator: "Exists"
    effect: "NoExecute"
  - operator: "Exists"
    effect: "NoSchedule"
```

Comment (k82cn) [reply cluster/addons/metadata-proxy/gce/metadata-proxy.yaml]: done

**Comment by mikedanese** [comment]
@krzyzacy is it possible to run alpha gci tests on this PR?

LGTM otherwise.

File: cluster/addons/metadata-proxy/gce/metadata-proxy.yaml (hunk 2 above)
Comment (liggitt) [reply cluster/addons/metadata-proxy/gce/metadata-proxy.yaml]: in updating the operators, the network unavailable component got dropped... please reverify the structure of the toleration

**Comment by jdumars** [comment]
@bsalamat PTAL ASAP!

**Comment by krzyzacy** [comment]
nope.. we can local run it, but it might have a different environment, anything is better than a borked job :-) I'd get this in and watch CI signals.

File: cluster/addons/metadata-proxy/gce/metadata-proxy.yaml (hunk 2 above)
Comment (liggitt) [reply cluster/addons/metadata-proxy/gce/metadata-proxy.yaml]: oh, I see... the intent was to tolerate any taint

Comment (k82cn) [reply cluster/addons/metadata-proxy/gce/metadata-proxy.yaml]: @liggitt , this suggests to tolerate everything.

**Comment by liggitt** [comment]
/hold cancel
as discussed in sig-release burndown

**Comment by bsalamat** [comment]
Late to the party. Thanks, @k82cn for the fix(es)!
